src/MongoDB.Analyzer/Core/TypesProcessor.cs

```diff
@@ -53,6 +56,11 @@ public string GetTypeSymbolToGeneratedTypeMapping(ITypeSymbol typeSymbol)
             return result.NewName;
         }
 
+        if (typeSymbol.IsSupportedSystemType())
```

Comment:
As this methods grows, probably we should extract it to
`private (string remappedName, string fullTypeName) GetGeneratedTypeMapping(...)`
and reuse it in both `GetTypeSymbolToGeneratedTypeMapping` and `ProcessTypeSymbol`

Reply:
Done